src/datetime/mod.rs

```diff
@@ -411,10 +411,8 @@ impl<Tz: TimeZone> DateTime<Tz> {
     ///   daylight saving time transition.
     #[must_use]
     pub fn checked_add_days(self, days: Days) -> Option<Self> {
-        self.naive_local()
-            .checked_add_days(days)?
-            .and_local_timezone(TimeZone::from_offset(&self.offset))
-            .single()
+        let naive = try_opt_ok!(self.naive_local().checked_add_days(days));
```

Comment:
The merge of main with 0.5.x is going to conflict here with the changes in #1333. But I can fix that in the merge PR.